velox/core/QueryCtx.h

```diff
@@ -78,12 +78,15 @@ class QueryCtx {
   }
 
   folly::Executor* executor() const {
+    VELOX_CHECK(isExecutorSupplied(), "Executor was not supplied.");
     if (executor_ != nullptr) {
       return executor_;
     }
-    auto executor = executorKeepalive_.get();
-    VELOX_CHECK(executor, "Executor was not supplied.");
-    return executor;
+    return executorKeepalive_.get();
+  }
+
+  bool isExecutorSupplied() const {
```

Comment:
Can this be a private method?

Reply:
It's exposed for being called from https://github.com/facebookincubator/velox/pull/8406/files#diff-49d6196eeda9e06d416ab94b2293fc9eed75045c441767764b8cff732177555aR206-R208 and https://github.com/facebookincubator/velox/pull/8406/files#diff-49d6196eeda9e06d416ab94b2293fc9eed75045c441767764b8cff732177555aR301-R303.